File: apps/web/src/features/shared/assignmentError.mjs
const driverMessages = {
  DRIVER_DEBT_LIMIT: 'Превышен лимит долга. Свяжитесь с оператором.',
  ORDER_REGION_MISMATCH: 'Заказ относится к другому региону.',
  DRIVER_PREVIOUSLY_CANCELLED_ORDER: 'Вы уже отменили этот заказ. Выберите другой.',
  DRIVER_BLOCKED_BY_CLIENT: 'Этот заказ недоступен для вас. Выберите другой.',
  CLIENT_BLOCKED_BY_DRIVER: 'Вы заблокировали этого пассажира. Выберите другой заказ.',
};

export function assignmentErrorMessage(code, driver = false) {
  if (!Object.hasOwn(driverMessages, code)) return null;
  // Do not disclose a driver's debt or another person's block to the rider.
  return driver ? driverMessages[code]
    : 'Водитель больше не может принять этот заказ. Выберите другого водителя или отклоните предложение.';
}

File: apps/mobile/smarttaxi_app/lib/features/shared/assignment_error.dart
import '../../l10n/app_localizations.dart';

String? assignmentErrorMessage(String? code, AppLocalizations l10n,
    {bool driver = false}) {
  final messages = {
    'DRIVER_DEBT_LIMIT': l10n.driverErrorDebtLimit,
    'ORDER_REGION_MISMATCH': l10n.driverErrorOrderRegionMismatch,
    'DRIVER_PREVIOUSLY_CANCELLED_ORDER': l10n.assignmentDriverCancelled,
    'DRIVER_BLOCKED_BY_CLIENT': l10n.assignmentDriverUnavailable,
    'CLIENT_BLOCKED_BY_DRIVER': l10n.assignmentDriverBlockedRider,
  };
  if (!messages.containsKey(code)) return null;
  // Financial details and another person's block are not passenger copy.
  return driver ? messages[code] : l10n.assignmentRiderUnavailable;
}
